src/main/java/org/tron/common/runtime/Runtime.java

```diff
@@ -266,10 +424,149 @@ public void go() {
+       *
+       * */
+    }
+    ContractCapsule contract = deposit.getContract(result.getContractAddress());
+    ByteString originAddress = contract.getInstance().getOriginAddress();
+    AccountCapsule origin = deposit.getAccount(originAddress.toByteArray());
+
+    byte[] callerAddressBytes = TransactionCapsule.getOwner(trx.getRawData().getContract(0));
+    AccountCapsule caller = deposit.getAccount(callerAddressBytes);
+
+    spendCpuUsage(cpuUsage, origin, caller);
+    if (spandStorage) {
+      spendStorageUsage(storageUsage, origin, caller);
+    }
+  }
+
+  private void spendCpuUsage(long cpuUsage, AccountCapsule origin, AccountCapsule caller) {
+    //TODO get origin cpu
+    long originCpu;
+    originCpu = getCpuByAccount(origin);
+
+    //TODO get caller cpu
+    long callerCpu;
+    callerCpu = getCpuByAccount(caller);
+    //TODO get caller cpulimit（trx）
+    long callerCpuLimit;
+    callerCpuLimit = 0;
+    cpuUsage = getCpuUsageLess(cpuUsage, origin, originCpu);
+    if (cpuUsage <= 0) {
+      return;
+    }
+    cpuUsage = getCpuUsageLess(cpuUsage, caller, callerCpu);
+    if (cpuUsage <= 0) {
+      return;
+    }
+    long overCpu = getCpuByLimit(callerCpuLimit);
+    cpuUsage = getCpuUsageLess(cpuUsage, caller, overCpu);
+    if (cpuUsage <= 0) {
+      return;
+    }
+  }
+
+  private long getCpuUsageLess(long cpuUsage, AccountCapsule origin, long cpu) {
+    if (cpuUsage <= cpu) {
+      //todo origin add cpu -cpuUsage
+    } else {
+      //todo origin add cpu -cpu
+    }
+    deposit.getDbManager().getAccountStore().put(origin.getAddress().toByteArray(), origin);
+    return cpuUsage - cpu;
+  }
+
+  private long getCpuByLimit(long callerCpuLimit) {
+
+    // TODO conversion limit(trx) to cpu
+    return 0;
+  }
+
+  private void spendStorageUsage(long storageUsage, AccountCapsule origin, AccountCapsule caller) {
+    //TODO get origin  storage
+    long orginStorage;
+    orginStorage = getStorageByAccount(origin);
+
+    //TODO get caller  storage
+    long callerStorage;
+    callerStorage = getStorageByAccount(caller);
+    //TODO get caller storagelimit（trx）
+    long callerStorageLimit;
+    callerStorageLimit = 0;
+    //storage 大于 1+2+3  不执行commit result 设置Error 正常退出 processesTransaction
+    storageUsage = getStorageUsageLess(storageUsage, origin, orginStorage);
+    if (storageUsage <= 0) {
+      return;
+    }
+    storageUsage = getStorageUsageLess(storageUsage, caller, callerStorage);
+    if (storageUsage <= 0) {
+      return;
+    }
+    long overStorage = getStorageByLimit(callerStorageLimit);
+    storageUsage = getStorageUsageLess(storageUsage, caller, overStorage);
+    if (storageUsage <= 0) {
+      return;
+    }
+  }
+
+  private long getStorageByLimit(long callerCpuLimit) {
+
+    // TODO conversion limit(trx) to cpu
+    return 0;
+  }
+
+  private long getStorageUsageLess(long storageUsage, AccountCapsule origin, long storage) {
+    if (storageUsage <= storage) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this conditional structure or edit its code blocks so that they're not all the same. ([squid:S3923](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS3923))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1188&issues=AWTgZvGN3lT_iufdZyBF&open=AWTgZvGN3lT_iufdZyBF)

[ISSUE_KEY]: AWTgZvGN3lT_iufdZyBF